File: 1337. The K Weakest Rows in a Matrix/solution.js
/**
 * @param {number[][]} mat
 * @param {number} k
 * @return {number[]}
 */
 var kWeakestRows = function(mat, k) {
  let maxRows = mat.length;
  let maxColumns = mat[0].length;

  result = [];

  for (let columns = 0; columns < maxColumns; columns++){
    for (let rows = 0; rows < maxRows; rows++){
      if (result.includes(rows)){
        continue;
      }
      if (mat[rows][columns] === 0){
        result.push(rows);
        if (result.length === k){
          return result;
        }
      }
    }
  }
  
  if (result.length !== k){
    for (let rows = 0; rows < maxRows; rows++){
      if (result.includes(rows)){
        continue;
      }
      result.push(rows);
      if (result.length === k){
        return result;
      }
    }
  }
};

File: 1337. The K Weakest Rows in a Matrix/solution.py
class Solution(object):
  def kWeakestRows(self, mat, k):
    """
    :type mat: List[List[int]]
    :type k: int
    :rtype: List[int]
    """
    maxRows = len(mat)
    maxColumns = len(mat[0])
    
    result = []
    
    for columns in range(maxColumns):
      for rows in range(maxRows):
        if rows in result:
          continue
        if mat[rows][columns] == 0:
          result.append(rows)
          if len(result) == k:
            return result
    
    if len(result) != k:
      for rows in range(maxRows):
        if rows in result:
          continue
        result.append(rows)
        if len(result) == k:
          return result